cypress/e2e/collaborators.spec.ts

```diff
@@ -0,0 +1,181 @@
+  cy.contains(E2E_EMAIL_TEST_SITE.repo).should("be.visible")
+}
+
+const getCollaboratorsModal = () => {
+  cy.contains("Site collaborators")
+    .parent()
+    .parent()
+    .parent()
+    .within(() => cy.get("button").click())
+
+  // NOTE: the form encloses the displayed modal component
+  return cy.get("form").should("be.visible")
+}
+
+const inputCollaborators = (user: string) => {
+  getCollaboratorsModal().get(ADD_COLLABORATOR_INPUT_SELECTOR).type(user).blur()
+  // NOTE: need to ignore the 422 w/ specific error message because we haven't ack yet
+  cy.contains("Add collaborator").click().wait(Interceptors.POST)
+}
+
+const removeCollaborator = (email: string) => {
+  cy.get("form")
+    .contains("div", email)
+    .parent()
+    .parent()
+    .within(() => {
+      cy.get('button[id^="delete-"]').click()
+    })
+
+  cy.contains("button", "Remove collaborator").click().wait(Interceptors.DELETE)
+}
+
+describe("collaborators flow", () => {
+  beforeEach(() => {
+    cy.setEmailSessionDefaults("Email admin")
+    cy.setupDefaultInterceptors()
+    visitE2eEmailTestRepo()
+  })
+
+  describe("Admin adding a collaborator", () => {
+    it("should not be able to click the add collaborator button when the input is empty", () => {
+      // Act
+      // Assert
+      getCollaboratorsModal().contains("Add collaborator").should("be.disabled")
+    })
+    it("should not be able to add a non-whitelisted collaborator", () => {
+      // Act
+      inputCollaborators("some_gibberish@gmail.com")
+
+      // Assert
+      cy.contains(ADD_COLLABORATOR_ERROR_MESSAGE).should("be.visible")
+    })
+    it("should not be able to add an existing user", () => {
+      // Act
+      inputCollaborators(E2E_EMAIL_ADMIN.email)
+      ignoreDuplicateError()
+
+      // Assert
+      cy.contains(DUPLICATE_COLLABORATOR_ERROR_MESSAGE).should("be.visible")
+    })
+    it("should not be able to add a collaborator without an isomer account", () => {
+      // Act
+      // NOTE: Initial admin will always be added manually
+      inputCollaborators("gibberish@nonsense.gov.sg")
+      ignoreNotFoundError()
+
+      // Assert
+      cy.contains(NON_EXISTING_USER_ERROR_MESSAGE).should("be.visible")
+    })
+    it("should be able to add a collaborator", () => {
+      // Arrange
+      // NOTE: Mock a login with a collaborator account;
+      // This creates the user account in our backend.
+      cy.createEmailUser(
+        collaborator,
+        USER_TYPES.Email.Collaborator,
+        USER_TYPES.Email.Admin
+      )
+
+      // Act
+      // NOTE: Initial admin will always be added manually
+      inputCollaborators(collaborator)
+      // NOTE: Cannot proceed without acknowledgement
+      cy.contains("Continue").should("be.disabled")
+      cy.get("input[name='isAcknowledged']").next().click()
+      cy.contains("Continue").click().wait(Interceptors.POST)
+      ignoreAcknowledgementError()
```

Comment:
why do we need this here btw?

Reply:
ps didn't write this down in the comments - this is because the error message is gotten from the _backend_. concretely, this means that we receive a network error, fe extracts the error message and shows it to the user. because of that, we need to ignore network errors so that cypress doesn't fail

Reply:
added a comment!